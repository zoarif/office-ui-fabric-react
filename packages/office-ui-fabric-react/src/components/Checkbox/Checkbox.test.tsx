import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Checkbox } from './index';

describe('Checkbox', () => {
  it('renders Checkbox correctly', () => {
    const component = renderer.create(
      <Checkbox
        label='Standard checkbox'
        ariaDescribedBy={ 'descriptionID' }
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});