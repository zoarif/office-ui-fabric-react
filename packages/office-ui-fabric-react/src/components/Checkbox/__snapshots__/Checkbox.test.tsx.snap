// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders Checkbox correctly 1`] = `
<button
  aria-checked={false}
  aria-describedby="descriptionID"
  aria-disabled={undefined}
  aria-label={undefined}
  aria-labelledby={undefined}
  aria-posinset={undefined}
  aria-setsize={undefined}
  className=
      ms-Checkbox
      is-enabled
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background-color: transparent;
        background: none;
        border: none;
        cursor: pointer;
        display: block;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
        outline: none;
        padding-bottom: 0;
        padding-left: 0;
        padding-right: 0;
        padding-top: 0;
        position: relative;
      }
      &::-moz-focus-inner {
        border: 0;
      }
      .ms-Fabric--isFocusVisible &:focus:after {
        border: 1px solid #ffffff;
        bottom: -1px;
        content: "";
        left: -1px;
        outline: 1px solid #666666;
        position: absolute;
        right: -1px;
        top: -1px;
        z-index: 1;
      }
      &:hover .ms-Checkbox-checkbox {
        border-color: #212121;
      }
      &:focus .ms-Checkbox-checkbox {
        border-color: #212121;
      }
      &:hover .ms-Checkbox-text {
        color: #000000;
      }
      &:focus .ms-Checkbox-text {
        color: #000000;
      }
  disabled={undefined}
  id="checkbox-0"
  name={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  role="checkbox"
  type="button"
>
  <label
    className=
        ms-Checkbox-label
        {
          align-items: center;
          cursor: pointer;
          display: inline-flex;
          margin-bottom: 0;
          margin-left: -4px;
          margin-right: -4px;
          margin-top: 0;
          position: relative;
          text-align: left;
        }
    htmlFor="checkbox-0"
  >
    <div
      className=
          ms-Checkbox-checkbox
          {
            align-items: center;
            border-color: #666666;
            border-style: solid;
            border-width: 1px;
            box-sizing: border-box;
            display: flex;
            flex-shrink: 0;
            height: 20px;
            justify-content: center;
            margin-bottom: 0;
            margin-left: 4px;
            margin-right: 4px;
            margin-top: 0;
            overflow: hidden;
            transition-duration: 200ms;
            transition-property: background, border, border-color;
            transition-timing-function: cubic-bezier(.4, 0, .23, 1);
            user-select: none;
            width: 20px;
          }
    >
      <i
        aria-hidden={true}
        className=

            {
              display: inline-block;
            }
            {
              color: #ffffff;
              opacity: 0;
            }
        data-icon-name="CheckMark"
        role="presentation"
      />
    </div>
    <span
      className=
          ms-Checkbox-text
          {
            color: #333333;
            font-size: 14px;
            margin-bottom: 0;
            margin-left: 4px;
            margin-right: 4px;
            margin-top: 0;
          }
    >
      Standard checkbox
    </span>
  </label>
</button>
`;
